### DISCUSSION TITLE
syntax error - invalid string: control character must be escaped; last read: '"ass<U+0000>'; expected string literal

### DISCUSSION THREAD

**Discussion by rahul233343** [nlohmann/json#2373] [Q&A] [answered]
Hello,

I am facing this issue as json std::string which i receive from server contains \u0000. while parsing i am getting exception. 

**terminate called after throwing an instance of 'nlohmann::detail::parse_error'
  what():  [json.exception.parse_error.101] parse error at 15429: syntax error - invalid string: control character must be escaped; last read: '"ass<U+0000>'; expected string literal**

One strange thing is that, the same query when queried from the other application does not give the parsing error. So i am not sure whether to raise the bug on server side.

Any lead would be appreciated.

**Comment by nlohmann** [comment, marked as answer]
A JSON string must not contain control characters unless they are escaped with `\` or via `\uxxxx`:

![](https://www.json.org/img/string.png)

That is, a null byte in a string must be escaped like `\u0000`. I think your server may be too lax on this - or the null byte is the symptom of a bigger issue.

**Reply by rahul233343** [reply]
But in other application with same version of json, same query result donot gives exception. 

Is by any chance i may inserting this \u0000 to the string? 
I am just doing curl and curl is writing result to my std::string variable in c

**"{\"balances\":[{\"nam\u0000e\":\"ABC\",\"free\":\"0.0000\u00000000\",\"lock\":\"0.00000000\"},{\"name\":\"XYZ\",\"free\":\"0.00000000\",\"lock\":\"0.00000000\"}],\"type\":[\"DEF\"]}\u0000"**

and if i have to use this result string as it is, what should i have to do?

**Reply by nlohmann** [reply]
This looks as if there is an issue with the buffer you fill with curl.

**Reply by rahul233343** [reply]
thanks, its working now
